src/tests/block-tests.ts

```diff
@@ -754,6 +755,139 @@ describe('block tests', () => {
     expect(fetch.status).not.toBe(200);
   });
 
+  test('blocks v2 cursor pagination', async () => {
```

Comment:
What would happen if there's a reorg in this case? Would a previously valid cursor return a 404? Perhaps a test for this is a good idea

Reply:
We _could_ still resolve the page in this case by allowing `canonical=false` in the cursor lookup, but then we'd have weird behavior like the result cursor being different (the canonical hash) than the given one. We are also not guaranteed to support non-canonical cursors in future changes. 

So yeah I think 404 is good idea. I'll add test.

Reply:
Done